checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java

```diff
@@ -437,6 +446,7 @@ public Void visitMethod(MethodTree node, Void p) {
     Void ret = super.visitMethod(node, p);
     currentMethods.removeFirst();
     effStack.removeFirst();
+    receiverType = previousReceiverType;
```

Comment:
Should this be in a finally block, or would an exception be fatal enough that there is no point?